prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -186,16 +193,25 @@ class ReportFunction(
                         payloadName
                     )
 
-                    workflowEngine.azureEventService.trackEvent(
-                        ReportReceivedEvent(
-                            report.id,
-                            sender,
-                            actionHistory.filterParameters(request),
-                            request.headers["x-forwarded-for"]?.split(",")?.first()
-                                ?: request.headers["x-azure-clientip"].toString(),
-                            request.headers["content-length"].toString()
+                    reportEventService.createReportEvent(
+                        eventName = ReportStreamEventName.REPORT_RECEIVED_EVENT,
+                        childReport = report,
+                        pipelineStepName = TaskAction.receive
+                    ) {
+                        params(
+                            mapOf(
+                                ReportStreamEventProperties.REQUEST_PARAMETERS
+                                    to actionHistory.filterParameters(request),
+                                ReportStreamEventProperties.SENDER_NAME to sender.fullName,
+                                ReportStreamEventProperties.SENDER_IP to (
+                                    request.headers["x-forwarded-for"]?.split(",")
```

Comment:
I feel like a helper method for this logic would clean this up since I've seen it in multiple places in our existing codebase.

```kotlin
fun getSenderIP(headers: Headers): String { 
  // ...
}
```